### ISSUE TITLE
Autogtp GUI

### ISSUE THREAD

**Issue by marcocalignano** [leela-zero/leela-zero#376]
I would like to know what people think and expecially you @gtp from the Autogtp GUI that I started.
Try it out from my branch /gui .

**Comment by marcocalignano** [comment]
For the moment it starts three threads so you can see three games. It need to get a starting mask where the number of thread can be choose. and then started.

**Comment by ljn917** [comment]
I think it can attract more users, especially windows users.

**Comment by evanroberts85** [comment]
Are you aware of LeelaSabaki?
https://lifein19x19.com/viewtopic.php?f=9&t=15302

**Comment by odeint** [comment]
Sabaki looks nice but is kind of bloated, an 80MB main executable and several large .dlls just to look at .sgf files basically? Not sure we want to bundle that with everything. Even the compressed installer is 35MB on Win and more than 50MB on linux. I actually wonder how that happens, probably frameworks and stuff.

**Comment by billyswong** [comment]
For small GUI we have 48.9KB LeelaWatcher :) https://github.com/fsparv/LeelaWatcher/

**Comment by marcocalignano** [comment]
Does this have multithreading support?

**Comment by FelixBrendel** [comment]
@marcocalignano It just parses the console output for one game at a time.

**Comment by marcocalignano** [comment]
@FelixBrendel I connect an object with all the thread that are running so I can display multiple game at the same time

**Comment by FelixBrendel** [comment]
Won't each game be slower if you play more games at the same time?

**Comment by ljn917** [comment]
I think it needs to be compatible with no gui environment, otherwise it is
difficult to run on server.

Another issue, boss (main.cpp) has to be allocated on heap, otherwise you
get memory double free on exit.

**Comment by marcocalignano** [comment]
@FelixBrendel no. One game uses only few % of GPU some GPU can run 4-6 game at the same time and The CPU slow the total only a little bit

**Comment by SebastienFaurite** [comment]
It depends on the usage of your GPU. With the current network size, 1 game use 60% of my GTX 1070. So you can run more games in parallel. 
Each game will be a little slower but overall you produce more game per hour.

**Comment by ljn917** [comment]
It depends on utilization of gpu.

On Dec 14, 2017 5:30 PM, "FelixBrendel" <notifications@github.com> wrote:

Won't each game be slower if you play more games at the same time?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/gcp/leela-zero/issues/376#issuecomment-351856144>, or mute
the thread

**Comment by Dorus** [comment]
Last time i tested, 1 instance gave me 4 games/hour, 2 instances gave 5 games/hour. So i got 20% more games out of it, but per game it got slower (24 mins/game instead of 15).

**Comment by sethtroisi** [comment]
>1year with no discussion, closing out.
